packages/tabs/src/TabTitle.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useRef, useEffect, SetStateAction } from 'react';
+import PropTypes from 'prop-types';
+
+interface TabTitleProps {
+  active: boolean;
+  children: Array<React.ReactElement>;
+  disabled?: boolean;
+  className?: string;
+  id?: string;
+  dataTabId: string;
+  onClick?: React.MouseEventHandler;
+  onKeyDown?: React.KeyboardEventHandler;
+  ariaControls?: string;
+  setFocusedState: React.Dispatch<SetStateAction<Array<string>>>;
+}
+
+function TabTitle({
+  active,
+  children,
+  disabled,
+  className,
+  id,
+  dataTabId,
+  onClick,
+  onKeyDown,
+  ariaControls,
+  setFocusedState,
+  ...rest
+}: TabTitleProps) {
+  const titleRef = useRef<HTMLLIElement>(null);
+
+  useEffect(() => {
+    if (titleRef && titleRef.current && active && !disabled) {
+      titleRef.current.focus();
+    }
+  }, [active]);
+
+  return (
+    <li
+      {...rest}
+      ref={titleRef}
+      className={className}
+      id={id}
+      role="tab"
+      data-tab-id={dataTabId}
+      onClick={onClick}
+      onKeyDown={onKeyDown}
+      aria-controls={ariaControls}
+      aria-selected={active}
+      aria-disabled={disabled}
+      tabIndex={active ? 0 : -1}
+      onBlur={() =>
```

Comment:
I'd prefer defining these using `useCallback` rather than inlining the functions.

Reply:
👍